consensus/cryptarchia-engine/src/time.rs

```diff
@@ -57,3 +84,73 @@ impl Add<u32> for Epoch {
         Epoch(self.0 + rhs)
     }
 }
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct EpochConfig {
+    // The stake distribution is always taken at the beginning of the previous epoch.
+    // This parameters controls how many slots to wait for it to be stabilized
+    // The value is computed as epoch_stake_distribution_stabilization * int(floor(k / f))
+    pub epoch_stake_distribution_stabilization: NonZero<u8>,
+    // This parameter controls how many slots we wait after the stake distribution
+    // snapshot has stabilized to take the nonce snapshot.
+    pub epoch_period_nonce_buffer: NonZero<u8>,
+    // This parameter controls how many slots we wait for the nonce snapshot to be considered
+    // stabilized
+    pub epoch_period_nonce_stabilization: NonZero<u8>,
+}
+
+impl EpochConfig {
+    pub fn epoch_length(&self, base_period_length: NonZero<u64>) -> u64 {
+        ((self.epoch_stake_distribution_stabilization.get()
```

Comment:
This can also overflow silently. But I guess we'll tackle them in a different PR.

Reply:
This was because of a mix of older version with the non-zero. Should be fine now.